Environment setting support.

tasdomas opened the pull request:
Added new config directive 'environment', that lists environment variables to be set on the tmux session. Needs tests and documentation, but feedback if the general approach seems sensible would be very useful.

tony commented on tmuxp/session.py:
```diff
@@ -330,6 +330,27 @@ def set_option(self, option, value):
+            if isinstance(proc.stderr, list) and len(proc.stderr) == int(1):
+                proc.stderr = proc.stderr[0]
+            raise ValueError('tmux set-environment stderr: %s' % proc.stderr)
+
```
thanks for the documentation

tony commented on tmuxp/testsuite/workspacebuilder.py:
```diff
@@ -186,7 +186,7 @@ def test_focus_pane_index(self):
 
         pane_path = '/usr'
-        for i in range(10):
+        for i in range(20):
```
Fine

tasdomas replied:
This test fails for me at a noticeable frequency with only 10 retries.

tony commented:
Good call on this @tasdomas 

I opened a question at https://github.com/tmux/tmux/issues/304 to double check if set-environment with ``-g`` is just server only.  If so we may want to chuck a ``set_environment`` to the server as well.

Tests / documentation / examples for this would be a good idea. It seems like a feature that'd be helpful in many situations.

tasdomas commented:
@tony, cool - I'll update the pull request and submit it for your consideration.

tony commented:
We probably should also add ``set-environment`` at the server level.  ``-g`` / "global" is the same as ``server``.

Any updates on this @tasdomas? If you're having trouble / time constraints let me know.

tasdomas commented:
@tony, working on it - currently trying to figure out a way to test the functionality. Once that's done, everything else is relatively easy. Should have a PR by next week - a little busy at the moment.

tony commented:
No problem, if you need help or need me to finish the test part, I'm happy to.

tasdomas commented:
Hello, @tony, I've updated the pull request. Extended the functionality to support setting global variables on the server as well, added some docs and enabled environment variable expansion for the variables (so setting something like PATH would be less tedious). Looking forward to your reviews.

tony commented:
@tasdomas: Pardon the delay. Merged.

This looks good to me and is useful enough of a feature to bump the version up.

Uploaded to pypi as 0.11.0.
